### PR TITLE
Geometry_oM: Explicit casts for Curves

### LINKED ISSUE #790: Geometry_oM: Polyline casts
#### Description:

As part of https://github.com/BHoM/BHoM/issues/789 and in support of https://github.com/BHoM/Dynamo_Toolkit/issues/224

Suggesting adding explicit casting:

- Polyline -> PolyCurve
- Polyline -> NurbsCurve

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR
Closes #790 
Think I'll leave #789 open as it also details step two, the casts with checks
related to #850 which has the wiki page

### Test files

https://burohappold.sharepoint.com/:u:/s/BHoM/EQBbBPE6BYFBlhJwbJllVxIB_ADwYP4S2qo4S03W9DWiwA?e=4BpRMQ

### Changelog

- `ICurve` to `PolyCurve`
- `Line`, `Polyline`, `Ellipse`, `Circle` to `NurbsCurve`
- `Circle` to `Ellipse`
- `Line` to `Polyline`

### Additional comments

There were mentions about various UI things in the issue #789 which I think was decided as not a problem. But could be worth double checking.

### CODE REVIEW

File: Geometry_oM/Curve/PolyCurve.cs [hunk 1]
```diff
@@ -34,7 +34,51 @@ public class PolyCurve : ICurve
...
+        {
+            return new PolyCurve() { Curves = new List<ICurve>() { curve } };
+        }
+
+        /***************************************************/
+
+        public static explicit operator PolyCurve(Polyline curve)
+        {
+            return new PolyCurve() { Curves = new List<ICurve>() { curve } };
+        }
```
Comment (pawelbaran) [inline Geometry_oM/Curve/PolyCurve.cs:80]: This should query `Polyline` for its segments first?

**Comment by kThorsager** [comment]
Yes, my plan was to implement all the unconditional casts first and follow up with those requiring checks.
Could do those now as well if you'd like?

File: Geometry_oM/Curve/PolyCurve.cs (hunk 1 above)
Comment (kThorsager) [reply Geometry_oM/Curve/PolyCurve.cs:80]: A mix of copy paste and lack of query methods in the oM, but can make it happen

**Comment by pawelbaran** [comment]
No opinion on the sequence, as long as we get this sorted :) happy to approve once my change request is resolved.

File: Geometry_oM/Curve/PolyCurve.cs (hunk 1 above)
Comment (pawelbaran) [reply Geometry_oM/Curve/PolyCurve.cs:80]: Yes please - I think the current implementation is going to do more harm than good:
image

:rofl:

**Comment by pawelbaran** [comment]
LGTM now, would be nice to get the missing ones sorted as soon as this is merged (to make sure this does not fall under the radar)

**Comment by al-fisher** [comment]
> Some are ofc a bit more comlex than one would hope to ahve in the oM, but given the functionality they provide I think well worth it, and it looks like you have done a good job of keeping the code to the minimum.

Agreed. 
Good job @kThorsager 👍

**Pull request merged**